### PR TITLE
[Auth0-js] Add missing onRedirecting property to CrossOriginLoginOptions interface

### PR DESCRIPTION
**Pull request by ihor-romanchuk**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/auth0/auth0.js/blob/master/src/web-auth/index.js
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Review by peterblazejewicz — COMMENTED** [review]
- which specific callback you're referring to? (the page linked lists 6 usages)
- inline docs seems to contradict definition (it accepts an argument):
https://github.com/auth0/auth0.js/blob/master/src/web-auth/index.js#L826
- add a test (it helps to review, Auth0 api has broad audience when used, e.g. I'm using mostly bearear based authentication
- follow established pattern, add ` | undefined;` it's here to support TS 4.4 strict optional propeerty types

**Comment by ihor-romanchuk** [comment]
Hi @peterblazejewicz, thank you for review. Sorry for the mess, this is my first contribution to the repo :)
- [WebAuth.prototype.login](https://github.com/auth0/auth0.js/blob/1a9e2a15d8bb7eb3fe993385ea4405a94487aac3/src/web-auth/index.js#L854) method accepts 'options' parameter. In type definitions this parameter is typed as `CrossOriginLoginOptions`. If you look into comments above method, you'll see [onRedirecting](https://github.com/auth0/auth0.js/blob/1a9e2a15d8bb7eb3fe993385ea4405a94487aac3/src/web-auth/index.js#L850) property. I've added type definitions for this property.
- `crossOriginLoginCallback` - it is a second parameter of login function. I've extended type definition for the first parameter.
- sure, will do my best
- I haven't heard about this TS feature yet, therefore though it's the same. Thanks will change.

It looks like type definitions are simply out of date. I've observed more discrepancies between type definitions and js implementation, but they are not so trivial. How should we handle package versioning? If we release new version to match the latest version of related js package, it will not be complete, there will be some problems with types.

File: types/auth0-js/index.d.ts [hunk 1]
```diff
@@ -350,7 +350,7 @@ export class Redirect {
             connection: string;
             /** allow userMetadata to be passed to signUp */
             userMetadata?: unknown | undefined;
-        },
+        } & CrossOriginLoginOptions,
```
Comment (ihor-romanchuk) [inline types/auth0-js/index.d.ts:353]: [Redirect.signupAndLogin](https://github.com/auth0/auth0.js/blob/1a9e2a15d8bb7eb3fe993385ea4405a94487aac3/src/web-auth/redirect.js#L55) passes `options` param down to [webauth.login](https://github.com/auth0/auth0.js/blob/1a9e2a15d8bb7eb3fe993385ea4405a94487aac3/src/web-auth/index.js#L854)

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@ihor-romanchuk thx!
sorry for the hassle on the first pr, but it is easy to get lost in Auth0 details (it's a large module, with different security topics, I'm using bearer token mostly only).
The CI failed for unrelated reason, if you could, push some kind of update to the PR)

**Review by peterblazejewicz — APPROVED** [review]

**Comment by ihor-romanchuk** [comment]
@peterblazejewicz do you know CI if failing? Could you help me to resolve it?

**Comment by peterblazejewicz** [comment]
Yes, there was issue in upstream packages (no related to DT), just rebase from master and force push changes to create another diff for CI

**Comment by ihor-romanchuk** [comment]
@peterblazejewicz CI is passing now, how can I merge this PR?

**Review by peterblazejewicz — APPROVED** [review]

**Comment by ihor-romanchuk** [comment]
Ready to merge

**Pull request merged**